### PR TITLE
lots of changes

### PR DESCRIPTION
**Pull request by jonathanjtan**

### CODE REVIEW

File: segment.py [hunk 1]
```diff
@@ -48,17 +50,19 @@ def lambda_handler(event, context):
...
-                destination = 'Segmented/{}'.format(filename)
-                s3_client.upload_file('/tmp/'+filename, bucket, destination)
+            for filename in os.listdir('/tmp/{}/{}*'.format(ConversionID, FilePath)):
+                if !filename.endswith('.mp3'):
+                    destination = 'Segmented/{}'.format(filename)
+                    s3_client.upload_file('/tmp/'+filename, bucket, destination)
             print "Uploading audio to s3"
             audiodestination = 'Audio/{}'.format(file_name)
-            s3_client.upload_file('/tmp/'+file_name+'.mp3', bucket, audiodestination)
+            s3_client.upload_file('{}.mp3'.format(FilePath), bucket, audiodestination)
```
Comment (gabinante) [inline segment.py]: We should upload to the same path we found the file originally, but in a folder. - A/B/C/D/ABCDOUEBOAWEUB/segment1.mp4

File: segment.py [hunk 2]
```diff
@@ -13,33 +11,37 @@ def lambda_handler(event, context):
     # Get the object from the event and show its content type
     # This job is triggered by FT_VideoConversions
 
-    Bucket = event[0]['dynamodb']['NewImage']['Bucket']
-    Key = event[0]['dynamodb']['NewImage']['Key']
-    ConversionID = event[0]['dynamodb']['NewImage']['ConversionID']
-    QueueMessageID = event[0]['dynamodb']['NewImage']['QueueMessageID']
+    Row = event[0]['dynamodb']['NewImage']
```
Comment (gabinante) [inline segment.py:14]: nice

File: segment.py [hunk 3]
```diff
@@ -13,33 +11,37 @@ def lambda_handler(event, context):
...
+    Bucket = Row['Bucket']
+    Filename = Row['Filename']
+    Path = Row['Path']
+    ConversionID = Row['ConversionID']
+    QueueMessageID = Row['QueueMessageID']
```
Comment (gabinante) [inline segment.py]: Not part of your code here, but We should prob consider renaming this to StatusQueueMessageID for clarity.

File: segment.py [hunk 4]
```diff
@@ -48,17 +50,19 @@ def lambda_handler(event, context):
...
             global destination
 
             # Each chunk is uploaded to s3
+            FilePath, Extension = os.path.splitext(LocalPath)
             print "Uploading segments to s3..."
-            for filename in os.listdir('/tmp/SEGMENT*'):
-                destination = 'Segmented/{}'.format(filename)
-                s3_client.upload_file('/tmp/'+filename, bucket, destination)
+            for filename in os.listdir('/tmp/{}/{}*'.format(ConversionID, FilePath)):
+                if !filename.endswith('.mp3'):
```
Comment (gabinante) [inline segment.py]: I think we can put mp3 in same directory as everything else, and remove this logic

File: segment.py [hunk 5]
```diff
@@ -13,33 +11,37 @@ def lambda_handler(event, context):
...
+    S3Path = "{}{}{}".format(Path, Filename, Extension)
+    LocalPath = "/tmp/{}/{}{}".format(ConversionID, Filename, Extension)
 
-    print "Bucket/Key is {}{}".format(Bucket, Key)
-    print "ConversionID is {}".format(ConversionID)
+    print "Bucket/ConversionID is {}, {}".format(Bucket, ConversionID)
+    print "StatusQueueMessageID is {}".format(StatusQueueMessageID)
 
-    if not Key.endswith('/'):
+    if not S3Path.endswith('/'):
```
Comment (jonathanjtan) [inline segment.py:26]: not sure if we need this?

File: segment.py [hunk 6]
```diff
@@ -1,6 +1,4 @@
-import boto3
-import ffmpy
-import os
+import boto3, ffmpy, os
 
 s3 = boto3.resource('s3')
 s3_client = boto3.client('s3')
```
Comment (jonathanjtan) [inline segment.py:4]: would also like to factor this out eventually

**Review by gabinante — APPROVED** [review]
LGTM

**Pull request merged**